### DISCUSSION TITLE
Error: Unsupported Node.js version: v20.12.2.

### DISCUSSION THREAD

**Discussion by speculative-artefact** [vercel/community#6983] [Help] [answered]
### Summary

hi, i'm struggling to deploy my sveltekit project. i'm getting the following error when deploying to Vercel

Build Failed
Command "npm run build" exited with 1

Error: Unsupported Node.js version: v20.12.2. Please use Node 16 or Node 18 to build your project, or explicitly specify a runtime in your adapter configuration.

i can build and preview locally with no errors
i have set my vercel project settings to use node v20.x
i have removed the defined engines node version in my package.json

what else should i look at?

### Steps to Reproduce

<img width="1208" alt="Screenshot 2024-05-26 at 10 40 23" src="https://github.com/vercel/community/assets/30245643/7830cbe1-c1b3-4b07-95ad-fc46850fadf2">

**Comment by speculative-artefact** [comment]
i've also tried to redeploy my last successful deployment but i get the same error

**Comment by swarnava** [comment, marked as answer]
Can you make sure you are using latest version of Vercel Adapter? Also do you have `engines` property set on `package.json` ?

**Reply by speculative-artefact** [reply]
thank you @swarnava , i updated all packages but noticed the vercel-adapter was not updating to the lastest version. i realised i needed to migrate my sveltekit and svelte to newer versions to be able to update the vercel adapter :/ the deployment is now successful!

**Reply by swarnava** [reply]
Good to know! Cheers! :)

**Comment by mishba-ai** [comment]
@doommchips I am also facing same errors as your were, did you update the vite,sveltjs and sveltekit??

**Reply by speculative-artefact** [reply]
hey, yes so what i did was migrate to the lastest version of sveltekit ([documentation](https://kit.svelte.dev/docs/migrating-to-sveltekit-2)), then the lastest version of svelte ([documentation](https://svelte.dev/docs/v4-migration-guide)). if i remember correctly i had to force one of them to update.

then you can run npm update and all your packages should auto update to the latest versions incl. vite. 

things might break so if you're running a local dev environment you'll have to stop it and start it again.

**Reply by mishba-ai** [reply]
Thank you @doommchips just successfully fixed it 👍🏼
